clients/js/cmds/sui.ts

```diff
@@ -74,64 +80,115 @@ exports.builder = function(y: typeof yargs) {
+          type: "string"
         })
         .option("governance-chain-id", {
           alias: "gci",
           describe: "Governance chain ID",
-          default: 1,
-          type: "number",
+          default: "3",
+          type: "string",
           required: false
         })
         .option("governance-contract", {
           alias: "gc",
           describe: "Governance contract",
           type: "string",
-          default: "0x0000000000000000000000000000000000000000000000000000000000000004",
+          default: "0000000000000000000000000000000000000000000000000000000000000004",
           required: false
         })
         .option("initial-guardian", {
           alias: "ig",
           required: true,
+          default: "beFA429d57cD18b7F8A4d91A2da9AB4AF05d0FBe",
           describe: "Initial guardian public keys)",
           type: "string",
         })
     }, async (argv) => {
       const network = argv.network.toUpperCase();
       assertNetwork(network);
       const rpc = argv.rpc ?? NETWORKS[network]["sui"].rpc;
+      const package_id = argv["package-id"]
       const deployer = argv.deployer;
       const chain_id = argv["chain-id"];
       const governance_chain_id = argv["governance-chain-id"];
       const governance_contract = argv["governance-contract"];
       const initial_guardian = argv["initial-guardian"];
       console.log("network: ", network)
       console.log("rpc: ", rpc)
-      console.log("deployer: ", deployer)
+      console.log("package id: ", package_id)
+      console.log("deployer object id: ", deployer)
       console.log("chain-id: ", chain_id)
       console.log("governance-chain-id: ", governance_chain_id)
       console.log("governance-contract: ", governance_contract)
       console.log("initial-guardian: ", initial_guardian)
+
       await callEntryFunc(
         network,
         rpc,
-        "packageObjectId",
+        String(package_id),
         "state",
         "init_and_share_state",
         [],
         [
           deployer,
           chain_id,
           governance_chain_id,
-          governance_contract,
-          [initial_guardian]
+          [...Buffer.from(governance_contract, "hex")],
+          [[...Buffer.from(initial_guardian, "hex")]]
         ],
     )
-      //await publishPackage(network, rpc, `${dir}/tokenbridge`);
     })
-    .command("init-coin", "Publish coin contract", (yargs) => {
+    .command("init-tokenbridge", "Init token bridge contract", (yargs) => {
+      return yargs
+        .option("network", network_options)
+        .option("rpc", rpc_description)
+        .option("package-id", {
+          alias: "pid",
+          describe: "Package/module ID",
+          required: true,
+          type: "string"
+        })
+        .option("deployer-capability", {
+          alias: "d",
+          describe: "Deployer capability object ID",
+          required: true,
+          type: "string",
+        })
+        .option("emitter-capability", {
```

Comment:
I just realised that the emitter capability is an argument to the init state function. any reason we can't create this object in the `init_and_share_state` function directly?

I propose:
1. remove the `emitter_cap` argument from `init_and_share_state` and create the object in there directly
2. remove the `get_new_emitter` entry function

Reply:
Actually, this doesn't save us an argument to the init state function, because we would have to pass in the `worm_state` to create a new emitter. Do you think one way is preferable to the other? I feel like they are about the same.

Reply:
ah good point, we'd need `worm_state`.

I feel like that's still a little bit better, because we don't need a separate step to create the emitter capability 🤔